workloads/files/workload-network-script-cm.yml

```diff
@@ -0,0 +1,426 @@
+  run.sh: |
+    #!/bin/sh
+    set -eo pipefail
+
+    log () {
+      echo "$(date -u) $1" 1>&2
+    }
+
+    # pbench Configuration
+    log "Configuring pbench for Network benchmarks"
+    mkdir -p /var/lib/pbench-agent/tools-default/
+    echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+    echo "" > /var/lib/pbench-agent/tools-default/oc
+    echo "workload" > /var/lib/pbench-agent/tools-default/label
+    if [[ -v ENABLE_PBENCH_AGENTS ]]; then
+      echo "" > /var/lib/pbench-agent/tools-default/disk
+      echo "" > /var/lib/pbench-agent/tools-default/iostat
+      echo "" > /var/lib/pbench-agent/tools-default/mpstat
+      echo "" > /var/lib/pbench-agent/tools-default/perf
+      echo "" > /var/lib/pbench-agent/tools-default/pidstat
+      master_nodes=`oc get nodes --show-labels -l pbench_agent=true,node-role.kubernetes.io/master= | awk '/master/ {print $1}'`
+      for node in $master_nodes; do
+        echo "master" > /var/lib/pbench-agent/tools-default/remote@$node
+      done
+      infra_nodes=`oc get nodes --show-labels -l pbench_agent=true,node-role.kubernetes.io/infra= | awk '/infra/ {print $1}'`
+      for node in $infra_nodes; do
+        echo "infra" > /var/lib/pbench-agent/tools-default/remote@$node
+      done
+      worker_nodes=`oc get nodes --show-labels -l pbench_agent=true,node-role.kubernetes.io/worker= | awk '/worker/ {print $1}'`
+      for node in $worker_nodes; do
+        echo "worker" > /var/lib/pbench-agent/tools-default/remote@$node
+      done
+    fi
+    source /opt/pbench-agent/profile
+    log "Done configuring pbench Network benchmarks"
+    # End pbench Configuration
+
+    # Start of Test Code
+    log "Running Network benchmarks"
+
+    # Template out the files
+    network_scc_yaml_template="$(cat /root/workload/network-scc.yaml.template)"
+    eval "echo \"${network_scc_yaml_template}\"" >> /tmp/network-scc.yaml
+    oc replace --force -f /tmp/network-scc.yaml
+
+    config_prefix="P2P"
+    if [[ "${NETWORK_TEST_HOSTNETWORK}" == "true" ]]; then
+      config_prefix="P2P_HN"
+    fi
+    # Loop over pair count
+    for pair_count in ${NETWORK_TEST_PAIRS//,/ }
+    do
+      log "Running Network benchmarks with ${pair_count} pair(s)"
+      network_yaml_template="$(cat /root/workload/p2p-network.yaml.template)"
+      if [ "${NETWORK_TEST_SERVICE}" == "true" ]; then
+        config_prefix="S2S"
+        network_yaml_template="$(cat /root/workload/s2s-network.yaml.template)"
+        cp /root/workload/svc-template.yaml /tmp/svc-template.yaml
+        # Append required ports to service template
+        for i in `seq 20000 20100`
+        do
+          echo -e "    - name: tcp-${i}\n      port: ${i}\n      protocol: TCP" >> /tmp/svc-template.yaml
+          echo -e "    - name: udp-${i}\n      port: ${i}\n      protocol: UDP" >> /tmp/svc-template.yaml
+        done
+      fi
+      eval "echo \"${network_yaml_template}\"" >> /tmp/network.yaml
```

Comment:
I'm still very uneasy about using eval for this.  It is a powerful friend, but a double-edged sword too.  In this case, you'll lose double quotes and '\\\\' will be evaluated as '\\'.  There might be other side-effects too I'm unaware of.  If this is intended and you know what you're doing, fine, but I'd personally use eval 1) only for environment dumps for reference what variables were used; 2) in cases where I know exactly what I'm doing and have a limited set of input for eval. @sjug, @chaitanyaenr feel free to chime in.

Reply:
I adjusted from `eval` to `envsubst` however we will need https://github.com/openshift-scale/images/pull/10 so the official workload container will have the command.  In the meantime you can use the one built in my namespace to test. (`quay.io/akrzos/scale-ci-workload`)

Reply:
No doubt `envsubst` is safer than eval, deferring to other reviewers if they think this is an ok design.